### PR TITLE
feat: @W-11665503 add support for liveTargetCallback

### PR DESCRIPTION
**Pull request by jdalton**
This allows Locker to tap into and control the liveness of a target.

### CODE REVIEW

File: packages/near-membrane-base/src/membrane.ts [hunk 1]
```diff
@@ -4212,82 +4233,15 @@ export function createMembraneMarshall(
...
-                          // https://tc39.es/ecma262/#sec-get-arraybuffer.prototype.bytelength
-                          // Step 2: Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
-                          try {
-                              if ('byteLength' in target) {
-                                  ReflectApply(ArrayBufferProtoByteLengthGetter!, target, []);
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                      }
```
Comment (jdalton) [inline packages/near-membrane-base/src/membrane.ts:4290]: ☝️ All of that goes is done in Locker anyways so now is handled by it. This reduces duplicate code and gives Locker the power to customize this check.

File: packages/near-membrane-base/src/membrane.ts [hunk 2]
```diff
@@ -4212,82 +4233,15 @@ export function createMembraneMarshall(
...
+                      try {
```
Comment (caridy) [inline packages/near-membrane-base/src/membrane.ts:4241]: do we really want to swallow this error?

File: packages/near-membrane-base/src/membrane.ts [hunk 3]
```diff
@@ -4212,82 +4233,15 @@ export function createMembraneMarshall(
                   }
                 : (noop as CallableInstallLazyPropertyDescriptors),
             // callableIsTargetLive
-            !IS_IN_SHADOW_REALM
-                ? (targetPointer: Pointer): boolean => {
+            !IS_IN_SHADOW_REALM && liveTargetCallback
```
Comment (caridy) [inline packages/near-membrane-base/src/membrane.ts:4236]: looking at this line, it looks like liveTarbgetCallback should we required. If it is not provided... then what?

File: packages/near-membrane-base/src/membrane.ts (hunk 2 above)
Comment (jdalton) [reply packages/near-membrane-base/src/membrane.ts:4241]: Yes. The contract there is a boolean result. If it errors while checking live-ness then it's likely not what we want to consider safe-to-be live.

**Comment by dgitin** [comment]
There seems to be a major difference between Chrome 104 and Chrome 105. I initially ran the tests on 104, after the update WeakSet is much faster. So whatever I have observed on Chrome 104 does not apply to Chrome 105.

**Comment by rwaldron** [comment]
This is great!

**Pull request merged**